lib/matplotlib/axes/_axes.py

```diff
@@ -4658,7 +4684,8 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
             alpha=alpha,
         )
         collection.set_transform(mtransforms.IdentityTransform())
-        if colors is None:
+        if ((colors is None or edgecolors is None) and not mcolors.is_color_like(c)
+                and not type(c) == np.ndarray):
```

Comment:
Another `type(...) == ...` that should be `isinstance`

Reply:
Updated